### PR TITLE
feat: Auto-focus on entity profile action modals

### PR DESCRIPTION
**Pull request by Salman-Apptware**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)- https://linear.app/acryl-data/issue/PRD-789/datahub-oss-auto-focus-on-entity-profile-action-modals
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/EditOwnersModal.tsx [hunk 1]
```diff
@@ -331,7 +332,18 @@ export const EditOwnersModal = ({
     function handleBlur() {
         setInputValue('');
     }
+    const [defaultValue, setDefaultValue] = useState<any>();
 
+    // If the modal content is dynamic or the modal is not rendered immediately.
+    // we ned need to use the useEffect hook to ensure that the Select component gets the default value after the modal is mounted.
+    useEffect(() => {
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/EditOwnersModal.tsx:339]: nice! do we need to do this anywhere else? tags, glossary terms?

File: datahub-web-react/src/app/entity/shared/containers/profile/utils.ts [hunk 2]
```diff
@@ -221,3 +221,6 @@ export function sortEntityProfileTabs(appConfig: AppConfig, entityType: EntityTy
 
     return sortedTabs;
 }
+
+// To focus out from modal popup so autofocus works
+export const getContainer = () => document.getElementById('root') as HTMLElement;
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/entity/shared/containers/profile/utils.ts:226]: again, did we try to add the id to the app.tsx? 

let's move this into a different place. let's put it in a root folder: src/app/utils/focus

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/containers/profile/utils.ts:226]: let's say getRootModalContainer. 

Also please write a bit of a more detailed comment here.

Comment (jjoyce0510) [reply datahub-web-react/src/app/entity/shared/containers/profile/utils.ts:226]: can we be 100% positive that the "root" id will never change?

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/EditOwnersModal.tsx (hunk 1 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/EditOwnersModal.tsx:339]: yes where we want default value we have to do this

File: datahub-web-react/src/app/entity/shared/containers/profile/utils.ts (hunk 2 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/shared/containers/profile/utils.ts:226]: `ReactDOM.render(
    <React.StrictMode>
        <App />
    </React.StrictMode>,
    document.getElementById('root'),
);`
here also we are using root as id if this cannot be change we are 100% sure id will not be change

Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/shared/containers/profile/utils.ts:226]: > let's move this into a different place. let's put it in a root folder: src/app/utils/focus
sure will add this to app/utils/focus and change name to `getModalDomContainer`

Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/shared/containers/profile/utils.ts:226]: > again, did we try to add the id to the app.tsx?

In App.tsx we have written all context and routing related code, for adding Id we have to add extra node of div and other person will be confused  why we have given id there instead we can use root as id that will not change

File: datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/EditOwnersModal.tsx (hunk 1 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/shared/containers/profile/sidebar/Ownership/EditOwnersModal.tsx:339]: in previous code we are using default value that's the reason I have added this code but in other place we are not using default value so not added this code there. In other places like glossary, term it is working expected without writing this code. If you want we can remove this code from `EditWonerModal` as well but in future we may required to add default value that's the reason I have kept this code here

File: datahub-web-react/src/app/entity/shared/containers/profile/utils.ts (hunk 2 above)
Comment (Salman-Apptware) [reply datahub-web-react/src/app/entity/shared/containers/profile/utils.ts]: update the MR please review and let me know

File: datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx [hunk 3]
```diff
@@ -92,6 +93,7 @@ export default function EditTagTermsModal({
     defaultValues = [],
     onOkOverride,
 }: EditTagsModalProps) {
+    console.log(visible,"visible=====")
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/shared/tags/AddTagsTermsModal.tsx:96]: feel free to get rid

File: smoke-test/tests/cypress/cypress/e2e/glossary/glossary_navigation.js [hunk 4]
```diff
@@ -30,6 +30,15 @@ describe("glossary sidebar navigation test", () => {
         cy.get('[data-testid="glossary-browser-sidebar"]').contains(glossaryTermGroup).click().wait(3000);
         cy.get('*[class^="GlossaryEntitiesList"]').contains(glossaryTerm).should("be.visible");
 
+        // create glossaryParentGroup
+        cy.goToGlossaryList();
```
Comment (jjoyce0510) [inline smoke-test/tests/cypress/cypress/e2e/glossary/glossary_navigation.js:34]: Can you explain why this is necessary? Seems that glossary_navigation is still not passing : /

**Comment by Salman-Apptware** [comment]
test failing so creating new branch

**Pull request closed without merging**